v0/destinations/yahoo_dsp/util.js

```diff
@@ -0,0 +1,280 @@
+const { CustomError, isDefinedAndNotNullAndNotEmpty } = require("../../util");
+
+const { ACCESS_TOKEN_CACHE_TTL, AUDIENCE_TYPE } = require("./config.js");
+const Cache = require("../../util/cache");
+
+const ACCESS_TOKEN_CACHE = new Cache(ACCESS_TOKEN_CACHE_TTL);
+
+const getUnixTimestamp = () => {
+  return Math.floor(Date.now() / 1000);
+};
+
+/**
+ * There are four location types which are stored in the poiLocationType error. These are ["chains", "woeids", "gids", "categories"].
+ * The ones to be added in includes Object will be taken from audienceType with include as Prefix. The ones to be added in excludes
+ * Object will be taken from audienceType with exclude as Prefix.
+ */
+const poiLocationType = [
+  "includeChains",
+  "includeWoeids",
+  "includeCategories",
+  "includeGids",
+  "excludeChains",
+  "excludeWoeids",
+  "excludeCategories",
+  "excludeGids"
+];
+
+/**
+ * This function is used to check if there is any common value in traits and poiLocationType array. This is done to check
+ * if any of the required location type is present in the input or not.
+ * @param {*} traits
+ * @param {*} poiLocationType
+ * @returns
+ */
+function isCommonElement(traits, poiLocationType) {
+  return traits.some(item => poiLocationType.includes(item));
+}
+
+/**
+ *
+ * @param {*} audienceList  - It is the list of audience to be added in the form of array". eg.
+ * [
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"},
+ *   {"email": "abc@email.com"}
+ * ]
+ * @param {*} Config
+ * @returns The function returns a hashed array of Audience List provided by the user like "email", "deviceId", "ipAddress".
+ * eg.[
+ * "251014dafc651f68edac7",
+ * "afbc34416ac6e7fbb9734",
+ * "42cbe7eebb412bbcd5b56",
+ * "379b4653a40878da7a584"
+ * ]
+ */
+const populateIdentifiers = (audienceList, Config) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  
+  if (isDefinedAndNotNullAndNotEmpty(audienceList)) {
+    // traversing through every userTraits in the add array for the traits to be added.
+    audienceList.forEach(userTraits => {
+      // storing keys of an object inside the add array.
+      const traits = Object.keys(userTraits);
+      // checking for the audience type the user wants to add is present in the input or not.
+      if (!traits.includes(AUDIENCE_TYPE[audienceType])) {
+        // throwing error if the audience type the user wants to add is not present in the input.
+        throw new CustomError(
+          `[Yahoo_DSP]:: Required property for ${AUDIENCE_TYPE[audienceType]} type audience is not available in an object`,
+          400
+        );
+      }
+      // here, hashing the data if is not hashed and pushing in the seedList array.
+      if (hashRequired) {
+        seedList.push(sha256(userTraits[AUDIENCE_TYPE[audienceType]]));
+      } else {
+        seedList.push(userTraits[AUDIENCE_TYPE[audienceType]]);
```

Comment:
this `AUDIENCE_TYPE[audienceType]` is redundant. Please declare a const variable and use that.

Reply:
done